modern/src/MainPage.js

```diff
@@ -36,39 +41,205 @@ const useStyles = makeStyles((theme) => ({
     },
     overflow: 'hidden',
   },
-  mapContainer: {
-    flexGrow: 1,
+  listContainer: {
+    position: 'absolute',
+    left: theme.spacing(1.5),
+    top: theme.spacing(10.5),
+    width: theme.dimensions.drawerWidthDesktop,
+    zIndex: 1301,
+    height: '100%',
+    maxHeight: `calc(100vh - ${theme.spacing(20)}px)`,
+    [theme.breakpoints.down('md')]: {
+      top: theme.spacing(7),
+      left: '0px',
+      width: '100%',
+      maxHeight: `calc(100vh - ${theme.spacing(14)}px)`,
+    },
+  },
+  paper: {
+    borderRadius: '0px',
+  },
+  toolbar: {
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+  },
+  deviceList: {
+    height: '100%',
+  },
+  collapsed: {
+    transform: `translateX(-${360 + 16}px)`,
+    transition: 'transform .5s ease',
+    [theme.breakpoints.down('md')]: {
+      transform: 'translateX(-100vw)',
+    },
+  },
+  uncollapsed: {
+    transform: `translateX(${theme.spacing(1.5)})`,
+    transition: 'transform .5s ease',
+    [theme.breakpoints.down('md')]: {
+      transform: 'translateX(0)',
+    },
+  },
+  deviceButton: {
+    position: 'absolute',
+    left: theme.spacing(1),
+    top: theme.spacing(10.5),
+    borderRadius: '0px',
+    [theme.breakpoints.down('md')]: {
+      left: theme.spacing(0),
+      top: theme.spacing(7),
+    },
+  },
+  deviceButtonBackground: {
+    backgroundColor: 'white',
+    color: '#777777',
+    '&:hover': {
+      backgroundColor: 'white',
+    },
+  },
+  bottomMenuContainer: {
+    position: 'absolute',
+    left: theme.spacing(1.5),
+    bottom: theme.spacing(1.5),
+    width: theme.dimensions.drawerWidthDesktop,
+    zIndex: 1301,
+    [theme.breakpoints.down('md')]: {
+      bottom: theme.spacing(0),
+      left: '0px',
+      width: '100%',
+    },
+  },
+  menuButton: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  iconText: {
+    fontSize: '0.75rem',
+    fontWeight: 'normal',
+    color: '#222222',
   },
 }));
 
-const MainPage = ({ width }) => {
+const MainPage = () => {
   const classes = useStyles();
+  const history = useHistory();
+  const theme = useTheme();
+
+  const [deviceName, setDeviceName] = useState();
+  const [collapsed, setCollapsed] = useState(false);
+
+  const matchesMD = useMediaQuery(theme.breakpoints.down('md'));
```

Comment:
Can we rename this to something more intuitive? Does this mean tablet or higher? Maybe we can have a variable called `isPhone` or something like that.

Reply:
This means tablet and lower. Renamed.